pkgs/development/libraries/mesa/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchurl, fetchpatch, buildPackages
+{ stdenv, lib, fetchurl, fetchpatch, buildPackages, pkgs
```

Comment:
Do we really need to use pkgs as an input?

Reply:
Do you mean introduce `pkgsCross` and `mesa` here to implement `tests.doesNotRelyOnTargetPlatform` below? That's would be more wired that introducing itself from arguments and also use `self` below.

Note that we cannot write `pkgsCross.aarch64-multiplatform.buildPackages.mesa == self` in the test. Otherwise overriding will definitely break the test. Since we doesn't have something like `makeDrvCrossAarch64 self == self`, my best approach is just the current diff.

Reply:
`pkgsCross` should basically never be used inside packages.

I wish it were in fact impossible (`pkgsCross` either not existing or slapped on after the fact) but there are a few odds and and and old grandfathered in things that I don't have the energy to fight.

Reply:
Hmm, then what's an alternative way to enforce target-invariance here?

Reply:
Is the current test acceptable? Or should I move the test into an individual one of nixosTests?

Reply:
Is there a problem with `pkgs` in the inputs?  I mean, we commonly pass `buildPackages` and similar, so is this really any different?